### PR TITLE
Optimizing calls to `Type_nod`

### PR DESCRIPTION
**Pull request by stefanhenneking**
The `element_data` function `Type_nod` is replaced by a parameterized array `TYPE_NOD` in `module node_types`. This will significantly reduce the time spent in `Type_nod` function calls which was previously called within loops where each function call required multiple `select case` comparisons.

@jbadger95 This should accelerate `elem_nodes_one` and `get_connect_info`, plus some other routines that require `Type_nod` info. If you do performance profiling, the time spent for getting `TYPE_NOD` info should now be insignificant compared to before.

### CODE REVIEW

**Review by jbadger95 — APPROVED** [review]
Awesome, great change!

**Review by stefanhenneking — COMMENTED** [review]
I'll wait until #102 is reviewed/merged, this PR is based on that branch.

**Review by jbadger95 — APPROVED** [review]
Great change!

**Pull request merged**
